explore/visualization_vega_lite_altair/Visualization_with_Vega-Lite_and_Altair.py

```diff
@@ -0,0 +1,238 @@
+    ('BiologicalProcess', 'PARTICIPATES_GpBP', 'GO:0035329'), # "hippo signaling" - Gene Ontology
+    ('BiologicalProcess', 'PARTICIPATES_GpBP', 'GO:0035330')  # "regulation of hippo signaling" - Gene Ontology
+]
+
+param_list = [dict(zip(names, qp)) for qp in query_params]
+
+
+# In[5]:
+
+query = '''
+MATCH (node)-[rel]-(gene)
+WHERE node.identifier = {node_id}
+  AND {label} in labels(node)
+  AND {rel_type} = type(rel)
+RETURN
+  gene.identifier as entrez_gene_id,
+  gene.name as gene_symbol
+ORDER BY gene_symbol
+'''
+
+
+# In[6]:
+
+driver = GraphDatabase.driver("bolt://neo4j.het.io")
+full_results_df = pd.DataFrame()
+with driver.session() as session:
+    for parameters in param_list:
+        result = session.run(query, parameters)
+        result_df = pd.DataFrame((x.values() for x in result), columns=result.keys())
+        full_results_df = full_results_df.append(result_df, ignore_index=True)
+
+classifier_genes_df = full_results_df.drop_duplicates().sort_values('gene_symbol').reset_index(drop=True)
+classifier_genes_df['entrez_gene_id'] = classifier_genes_df['entrez_gene_id'].astype('str')
+
+
+# In[7]:
+
+# Here are the genes that participate in the Hippo signaling pathway
+classifier_genes_df
+
+
+# #### Load Data
+
+# In[8]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('../download', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[9]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('../download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[10]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('../download', 'samples.tsv')\nclinical = pd.read_table(path, index_col=0)")
+
+
+# In[11]:
+
+# Subset the Y matrix to only the genes to be classified
+y_full = Y[classifier_genes_df['entrez_gene_id']]
+
+
+# In[12]:
+
+y_full.columns = classifier_genes_df['gene_symbol']
+y_full = y_full.assign(disease = clinical['disease'])
+
+# This matrix now stores the final y matrix for the classifier (y['indicator'])
+y = y_full.assign(indicator = y_full.max(axis=1))
+
+
+# In[13]:
+
+unique_pos = y.groupby('disease').apply(lambda x: x['indicator'].sum())
+heatmap_df0 = y_full.groupby('disease').sum().assign(TOTAL = unique_pos)
+heatmap_df = heatmap_df0.divide(y_full.disease.value_counts(sort=False).sort_index(), axis=0)
+
+
+# In[14]:
+
+# What is the percentage of different mutations across different cancer types?
+sns.heatmap(heatmap_df);
+
+
+# ### Part 2: Building a Heatmap using Vega-Lite/Altair API
+# 
+# In order to take advantage of the Altair API, we have to covert the data into the so called long format (often referred to as the tidy format).
+
+# In[15]:
+
+# Stack it: 
+heatmap_df_stacked = heatmap_df.stack()
+heatmap_df_stacked.head()
+
+
+# In[16]:
+
+# Convert into the Pandas dataframe
+df000 = pd.DataFrame(heatmap_df_stacked)
```

Comment:
Let's try to use only descriptive variable names -- no `000`. In instances where you are improving the dataframe, you can overwrite the same variable. For example, keep overwriting `heatmap_df_stacked`. In fact, you may be able to chain all the operations together, so you only have to assign to a variable once.